expr/func_groupbytags_test.go

```diff
@@ -13,54 +13,23 @@ import (
 )
 
 func getModel(name string, data []schema.Point) models.Series {
-	tags := make(map[string]string, 3)
-	tagSplits := strings.Split(name, ";")
-
-	tags["name"] = tagSplits[0]
-
-	for _, split := range tagSplits[1:] {
-		pair := strings.SplitN(split, "=", 2)
-		tags[pair[0]] = pair[1]
-	}
-	return models.Series{
+	series := models.Series{
 		Target:     name,
 		QueryPatt:  name,
-		Tags:       tags,
 		Datapoints: getCopy(data),
 	}
+	series.SetTags()
+	return series
 }
 
 // Test error cases
 func TestNoTags(t *testing.T) {
 	in := []models.Series{
 		getModel("name1;tag1=val1", a),
 	}
-	f := NewGroupByTags()
-	gby := f.(*FuncGroupByTags)
-	gby.in = NewMock(in)
-	gby.aggregator = "sum"
-	gby.tags = []string{}
-
-	_, err := f.Exec(make(map[Req][]models.Series))
-	if err == nil {
-		t.Fatalf("case %q: err should not be nil but was", "TestnoTags")
-	}
-}
-
-func TestInvalidAggregator(t *testing.T) {
```

Comment:
why is this one removed?

Reply:
The check was moved into the parse validator per your suggestion. There is no longer any checking in Exec for agg validity.

I could add it back for robustness/sanity.

Reply:
we could add it back, and adjust testGroupByTags to test more of the aggregator lifecycle rather than just exec, but that probably becomes too much hassle.
let's not worry about it for now, maybe later we can come up with an elegant way to test the validors (ideally in a more general way), but maybe we don't have to, cause it's also fairly obvious stuff.